src/workshopselector/WorkshopSelector.java

```diff
@@ -6,65 +6,80 @@
  */
 package workshopselector;
 
-import java.awt.Color;
-import java.awt.FlowLayout;
 import javax.swing.*;
-import javax.swing.JFrame;
-import javax.swing.JLabel;
-import javax.swing.JList;
-import javax.swing.JScrollPane;
-import javax.swing.ListSelectionModel;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 
 
- 
-public class WorkshopSelector {
+public class WorkshopSelector implements ActionListener {
+
+    private JList listWorkshops;
+    private JList listLocation;
+    private JLabel totalCostLabel;
+    private int[] allLocationFees;
+    private int[] allWorkshopFees;
+    private int[] allWorkshopDurations;
 
     /**
      * @param args the command line arguments
      */
     public static void main(String[] args) {
+        new WorkshopSelector().build();
+    } //end main()
+
+    private void build() {
+        //Setup costs/durations
+        allWorkshopFees = new int[]{800,1500, 1300};
+        allWorkshopDurations = new int[]{2, 3, 5};
+        allLocationFees = new int[]{225, 175, 300};
+
         //create the frame and the layout
         JFrame myJFrame = new JFrame();
         FlowLayout myLayout = new FlowLayout();
-        myLayout.setHgap(10);    
+        myLayout.setHgap(10);
         myLayout.setVgap(20);
 
         //add the layout to the frame
         myJFrame.setLayout(myLayout);
 
         //create components
         JLabel wksLabel = new JLabel("Select a workshop");
-        String[] workshops = new String[]{"Time Management", "Supervision Skills",
-            "Negotiation"};
-        JList listWorkshops = new JList(workshops);
+        String[] workshops = new String[]{"Time Management", "Supervision Skills", "Negotiation"};
+        listWorkshops = new JList(workshops);
         listWorkshops.setFixedCellHeight(20);
-    listWorkshops.setFixedCellWidth(130);
-        
+        listWorkshops.setFixedCellWidth(130);
+
         JLabel locLabel = new JLabel("Select a location");
         String[] locations = new String[]{"Chicago, IL", "Dallas, TX", "Orlando, FL"};
-        JList listLocation = new JList(locations);
+        listLocation = new JList(locations);
         listLocation.setFixedCellHeight(20);
         listLocation.setFixedCellWidth(130);
-        
-        
+
+
         //JLabel calcLabel = new JLabel();
         JButton  calcButton = new JButton("Calculate");
         calcButton.setForeground(Color.DARK_GRAY);
-        
-        //add the components to the layout 
+        calcButton.addActionListener(this);
+
+        totalCostLabel = new JLabel("Total Cost");
+
+        //add the components to the layout
         myJFrame.add(wksLabel);
         myJFrame.add(listWorkshops);
-       
+
         myJFrame.add(locLabel);
         myJFrame.add(listLocation);
-        
+
         myJFrame.add(calcButton);
-        
+        myJFrame.add(new JLabel(""));
```

Comment:
Ignore this wart. I was trying to line up the label under the button in a rush. (I suck at Swing!)